Presentation.Web/Infrastructure/APIRequestsLoggingMiddleware.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Web.Security;
+using Microsoft.Owin;
+using Ninject;
+using OwinNinjectExample.Ninject;
+using Serilog;
+
+namespace Presentation.Web.Infrastructure
+{
+    public class ApiRequestsLoggingMiddleware : OwinMiddleware
+    {
+
+        private ILogger _logger = Log.Logger;
+
+        public ApiRequestsLoggingMiddleware(OwinMiddleware next) : base(next)
+        {
+        }
+
+        public override async Task Invoke(IOwinContext context)
+        {
+            _logger = context.GetResolutionScope().Get<ILogger>();
+            if (context.Request.Headers.ContainsKey("Authorization"))
+            {
+                var startTime = DateTime.Now;
+                var route = context.Request.Path;
+                var method = context.Request.Method;
+                var queryParameters = getQueryParameters(context.Request.Query);
+                var userID = context.Request.User.Identity.Name;
+                await Next.Invoke(context);
+                var processingTime = DateTime.Now - startTime;
```

Comment:
**Ikke processing time**... ProcessingStart og ProcessingEnd
og husk DateTime.UTCNow i stedet for Now :-)